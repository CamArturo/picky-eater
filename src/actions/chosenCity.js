export const addChosenCity = (city) => {
  return {
    type: 'ADD_CHOSEN_CITY',
    city
  }
};