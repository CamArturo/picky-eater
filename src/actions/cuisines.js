export const sendCuisinesToStore = (cuisines) => {
  return {
    type: 'ADD_AVAILABLE_CUISINES',
    cuisines
  };
};