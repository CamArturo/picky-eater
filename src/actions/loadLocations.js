export const sendLocationsToStore = (locations) => (
  {
    type: 'STORE_CITIES',
    locations
  }
);