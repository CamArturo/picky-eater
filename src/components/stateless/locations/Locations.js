import React from 'react';

const Locations = () => {
  return (
    <section className="locations-container">
      <button>Newton, TX</button>
      <button>Newton, TX</button>
      <button>Newton, TX</button>
      <button>Newton, TX</button>
      <button>Newton, TX</button>
      <button>Newton, TX</button>
    </section>
  );
};

export default Locations;
