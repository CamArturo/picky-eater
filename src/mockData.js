export const chosenCity = {
  city: 'Newton, IL',
  id: 5172
};

export const mockLocations = [
  {
    'id': 7171,
    'name': 'Newton Grove, NC',
    'country_id': 216,
    'country_name': 'United States',
    'country_flag_url': 'https://b.zmtcdn.com/images/countries/flags/country_216.png',
    'should_experiment_with': 0,
    'discovery_enabled': 0,
    'has_new_ad_format': 0,
    'is_state': 0,
    'state_id': 95,
    'state_name': 'North Carolina',
    'state_code': 'NC'
  },
  {
    'id': 5172,
    'name': 'Newton, IL',
    'country_id': 216,
    'country_name': 'United States',
    'country_flag_url': 'https://b.zmtcdn.com/images/countries/flags/country_216.png',
    'should_experiment_with': 0,
    'discovery_enabled': 0,
    'has_new_ad_format': 0,
    'is_state': 0,
    'state_id': 82,
    'state_name': 'Illinois',
    'state_code': 'IL'
  },
  {
    'id': 6946,
    'name': 'Newton, MS',
    'country_id': 216,
    'country_name': 'United States',
    'country_flag_url': 'https://b.zmtcdn.com/images/countries/flags/country_216.png',
    'should_experiment_with': 0,
    'discovery_enabled': 0,
    'has_new_ad_format': 0,
    'is_state': 0,
    'state_id': 93,
    'state_name': 'Mississippi',
    'state_code': 'MS'
  }
];

export const availableCuisines = [
  {
    'cuisine': {
      'cuisine_id': 6,
      'cuisine_name': 'Afghani'
    }
  },
  {
    'cuisine': {
      'cuisine_id': 152,
      'cuisine_name': 'African'
    }
  },
  {
    'cuisine': {
      'cuisine_id': 1,
      'cuisine_name': 'American'
    }
  }
];

export const mockRestaurants = {
  restaurants:
    {
      results_found: 7,
      results_start: 0,
      results_shown: 7,
      restaurants:
        [[Object],
          [Object],
          [Object],
          [Object],
          [Object],
          [Object],
          [Object]]
    }
};

export const mockListRestaurants = [
  'Yesterday\'s Pub & Dining',
  'Olde Tyme Steakhaus',
  'Brew City Pub & Grill',
  'Holiday Restaurant',
  'Richard\'s Farm Restaurant',
  'Lone Star Steakhouse & Saloon',
  'Gray\'s'
];