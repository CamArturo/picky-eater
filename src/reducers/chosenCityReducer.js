export const chosenCity = (state = {}, action) => {
  switch (action.type) {
    case 'ADD_CHOSEN_CITY':
      return {
        city: action.city,
        id: action.id
      };
    default:
      return state
  }
};
