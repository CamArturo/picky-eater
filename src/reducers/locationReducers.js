export const locations = (state = [], action) => {
  switch (action.type) {
    case 'STORE_CITIES':
      return [...action.locations];
    default:
      return state
  }
};
