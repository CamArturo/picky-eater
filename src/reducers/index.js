import { combineReducers } from 'redux';
import { locations } from './locationReducers';
import { chosenCity } from './chosenCityReducer';

const rootReducer = combineReducers(
  {
    locations,
    chosenCity
  }
);

export default rootReducer;