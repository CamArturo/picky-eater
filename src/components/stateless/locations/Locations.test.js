// returning an object that has proper object with key and value of a function
// => aka the expected object.

// dispatch is being called with correct action
import { shallow, mount, render } from 'enzyme';
import { Locations } from './Locations';
import React from 'react';
import { mockLocations } from '../../../mockData';
import { mapStateToProps, mapDispatchToProps } from './Locations';
import { addChosenCity } from '../../../actions/chosenCity';
import { locations } from '../../../reducers/locationReducers';
import { chosenCity } from '../../../reducers/chosenCityReducer';

describe('Locations', () => {
  let wrapper;
  let mockState;
  beforeEach(() => {
    wrapper = shallow(<Locations locations={mockLocations} storeChosenCity={jest.fn} />);
    mockState = {
      locations: mockLocations,
      chosenCity: 'Newton, IL'
    };
  });
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should mapStateToProps', () => {
    const actual = mapStateToProps(mockState);
    const expected = {
      locations: mockLocations
    };
    expect(actual).toEqual(expected);
  });
  it('calls dispatch with an addChosenCity when a button is clicked', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = {"city": "Newton, IL", "type": "ADD_CHOSEN_CITY"};

    // const actionToDispatch = addChosenCity('Newton, IL');
    const mappedProps = mapDispatchToProps(mockDispatch);

    mappedProps.storeChosenCity('Newton, IL');

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  });
  it('should fire storeChosenCity when a button is clicked', () => {

  });
  it('should only render buttons when there are more than 0 locations in store', () => {

  });
});