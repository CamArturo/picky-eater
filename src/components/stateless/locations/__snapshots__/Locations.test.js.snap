// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Locations should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Locations
    locations={
        Array [
            Object {
              "country_flag_url": "https://b.zmtcdn.com/images/countries/flags/country_216.png",
              "country_id": 216,
              "country_name": "United States",
              "discovery_enabled": 0,
              "has_new_ad_format": 0,
              "id": 7171,
              "is_state": 0,
              "name": "Newton Grove, NC",
              "should_experiment_with": 0,
              "state_code": "NC",
              "state_id": 95,
              "state_name": "North Carolina",
            },
            Object {
              "country_flag_url": "https://b.zmtcdn.com/images/countries/flags/country_216.png",
              "country_id": 216,
              "country_name": "United States",
              "discovery_enabled": 0,
              "has_new_ad_format": 0,
              "id": 5172,
              "is_state": 0,
              "name": "Newton, IL",
              "should_experiment_with": 0,
              "state_code": "IL",
              "state_id": 82,
              "state_name": "Illinois",
            },
            Object {
              "country_flag_url": "https://b.zmtcdn.com/images/countries/flags/country_216.png",
              "country_id": 216,
              "country_name": "United States",
              "discovery_enabled": 0,
              "has_new_ad_format": 0,
              "id": 6946,
              "is_state": 0,
              "name": "Newton, MS",
              "should_experiment_with": 0,
              "state_code": "MS",
              "state_id": 93,
              "state_name": "Mississippi",
            },
          ]
    }
    storeChosenCity={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          onClick={[Function]}
>
          Newton Grove, NC
</button>,
        <button
          onClick={[Function]}
>
          Newton, IL
</button>,
        <button
          onClick={[Function]}
>
          Newton, MS
</button>,
      ],
      "className": "locations-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "key0",
        "nodeType": "host",
        "props": Object {
          "children": "Newton Grove, NC",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Newton Grove, NC",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "key1",
        "nodeType": "host",
        "props": Object {
          "children": "Newton, IL",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Newton, IL",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "key2",
        "nodeType": "host",
        "props": Object {
          "children": "Newton, MS",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Newton, MS",
        "type": "button",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            onClick={[Function]}
>
            Newton Grove, NC
</button>,
          <button
            onClick={[Function]}
>
            Newton, IL
</button>,
          <button
            onClick={[Function]}
>
            Newton, MS
</button>,
        ],
        "className": "locations-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "key0",
          "nodeType": "host",
          "props": Object {
            "children": "Newton Grove, NC",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Newton Grove, NC",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "key1",
          "nodeType": "host",
          "props": Object {
            "children": "Newton, IL",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Newton, IL",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "key2",
          "nodeType": "host",
          "props": Object {
            "children": "Newton, MS",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Newton, MS",
          "type": "button",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
